qt/widgets/common/src/AlgorithmHistoryWindow.cpp

```diff
@@ -696,6 +736,28 @@ void AlgHistoryTreeWidget::uncheckAllChildren(QTreeWidgetItem *item,
   }
 }
 
+void AlgHistoryTreeWidget::removeHiddenChildren(QTreeWidgetItem *item) {
+  if (auto parent = item->parent()) {
+    int index{parent->indexOfChild(item)};
+
+    for (int i = 0; i < item->childCount(); i++) {
+      if (item->child(i)->checkState(1) == Qt::Checked) {
+        removeHiddenChildren(item->child(i));
+      }
+      delete parent->child(index + 1);
+    }
+  } else {
+    int index{indexOfTopLevelItem(item)};
+
+    for (int i = 0; i < item->childCount(); ++i) {
+      if (item->child(i)->checkState(index) == Qt::Checked) {
```

Comment:
Why is this `checkState(index)` when above it is `checkState(1)`?

Reply:
That was a mistake, it should be `checkState(1)` because it is referring to the columns in the tree widget and the second column is the one with the check boxes. I've fixed this now.